scripts/check_links.py

```diff
@@ -1,57 +1,83 @@
 class MatchTuple(NamedTuple):
     source: str
     name: str
     link: str
 
-def url_ok(match_tuple: MatchTuple) -> Tuple[MatchTuple, bool]:
+
+def url_ok(match_tuple: MatchTuple) -> bool:
+    """Check if a URL is reachable."""
     try:
-        result = requests.get(match_tuple.link)
-        print(f"  {result.status_code}: {match_tuple.source}")
-        return (match_tuple, result.ok)
-    except requests.ConnectionError:
-        return (match_tuple, False)
+        result = requests.get(match_tuple.link, timeout=5)
+        return result.ok
+    except (requests.ConnectionError, requests.Timeout):
+        return False
+
+
+def path_ok(match_tuple: MatchTuple) -> bool:
+    """Check if a file in this repository exists."""
+    relative_path = match_tuple.link.split("#")[0]
+    full_path = os.path.join(os.path.dirname(str(match_tuple.source)), relative_path)
+    return os.path.exists(full_path)
 
-if __name__ == "__main__":
 
+def link_ok(match_tuple: MatchTuple) -> Tuple[MatchTuple, bool]:
+    if match_tuple.link.startswith("http"):
+        result_ok = url_ok(match_tuple)
+    else:
+        result_ok = path_ok(match_tuple)
+    print(f"  {'✓' if result_ok else '✗'} {match_tuple.link}")
+    return match_tuple, result_ok
+
+
+def main():
     print("Finding all markdown files in the current directory...")
 
     project_root = (pathlib.Path(__file__).parent / "..").resolve() # pylint: disable=no-member
     markdown_files = project_root.glob('**/*.md')
 
     all_matches = set()
-    url_regex = re.compile(r'\[([^!][^\]]+)\]\((http[s]?[^)(]+)\)')
+    url_regex = re.compile(r'\[([^!][^\]]+)\]\(([^)(]+)\)')
     for markdown_file in markdown_files:
         with open(markdown_file) as handle:
             for line in handle.readlines():
                 matches = url_regex.findall(line)
                 for name, link in matches:
                     if 'localhost' not in link:
                         all_matches.add(MatchTuple(source=str(markdown_file), name=name, link=link))
-    print(f"  {len(all_matches)} markdown files found")
 
-    print("Checking to make sure we can retrieve each web URL...")
+    print(f"  {len(all_matches)} markdown files found")
+    print("Checking to make sure we can retrieve each link...")
 
     with Pool(processes=10) as pool:
-        results = pool.map(url_ok, [match for match in list(all_matches)])
+        results = pool.map(link_ok, [match for match in list(all_matches)])
     unreachable_results = [result for result in results if not result[1]]
 
     if unreachable_results:
-        print(f"Unreachable Links ({len(unreachable_results)}):")
-        for index, result in enumerate(unreachable_results):
-            print("  " + str(index))
-            print("  Source: " + result[0].source)
-            print("  Name: " + result[0].name)
-            print("  Link: " + result[0].link)
+        print(f"Unreachable links ({len(unreachable_results)}):")
+        for result in unreachable_results:
```

Comment:
Minor nit: `for match_tuple, _ in unreachable_results:` would be a bit better than doing `result[0]` below, just because then `result[0]` has a semantically meaningful name.

Reply:
Good point!